packages/apollo-server-integration-testsuite/src/ApolloServer.ts

```diff
@@ -240,6 +276,68 @@ export function testApolloServer<AS extends ApolloServerBase>(
 
           process.env.NODE_ENV = nodeEnv;
         });
+
+        it('prohibits providing a gateway in addition to schema/typedefs/resolvers', async () => {
+          const { gateway } = makeGatewayMock();
+
+          const incompatibleArgsSpy = jest.fn();
+          await createApolloServer({ gateway, schema }).catch(err =>
+            incompatibleArgsSpy(err.message),
+          );
+          expect(incompatibleArgsSpy.mock.calls[0][0]).toMatch(
+            /Cannot define both/,
+          );
+
+          await createApolloServer({ gateway, modules: {} as any }).catch(err =>
+            incompatibleArgsSpy(err.message),
+          );
+          expect(incompatibleArgsSpy.mock.calls[1][0]).toMatch(
+            /Cannot define both/,
+          );
+
+          await createApolloServer({ gateway, typeDefs: {} as any }).catch(
+            err => incompatibleArgsSpy(err.message),
+          );
+          expect(incompatibleArgsSpy.mock.calls[2][0]).toMatch(
+            /Cannot define both/,
+          );
+        });
+
+        it('prohibits providing a gateway in addition to subscription options', async () => {
+          const { gateway } = makeGatewayMock();
+
+          const incompatibleArgsSpy = jest.fn();
+          await createApolloServer({
+            gateway,
+            subscriptions: 'pathToSubscriptions',
+          }).catch(err => incompatibleArgsSpy(err.message));
+          expect(incompatibleArgsSpy.mock.calls[0][0]).toMatch(
+            /Cannot define both `subscriptions` and `gateway`./,
+          );
+
+          await createApolloServer({
+            gateway,
+            subscriptions: true as any,
+          }).catch(err => incompatibleArgsSpy(err.message));
+          expect(incompatibleArgsSpy.mock.calls[1][0]).toMatch(
+            /Cannot define both `subscriptions` and `gateway`./,
+          );
+
+          await createApolloServer({
+            gateway,
+            subscriptions: { path: '' } as any,
+          }).catch(err => incompatibleArgsSpy(err.message));
+          expect(incompatibleArgsSpy.mock.calls[2][0]).toMatch(
+            /Cannot define both `subscriptions` and `gateway`./,
+          );
+
+          await createApolloServer({
+            gateway,
+          }).catch(err => incompatibleArgsSpy(err.message));
+          expect(incompatibleArgsSpy.mock.calls[3][0]).toMatch(
+            /Cannot define both `subscriptions` and `gateway`./,
```

Comment:
Nit: I could see:

```
const expectedError = /Cannot define both `subscriptions` and `gateway`./;
```

...being a good candidate to DRY this up a bit (especially since changing the error message would necessitate changing each of these).